torch/fx/passes/shape_prop.py

```diff
@@ -28,10 +30,43 @@ class TensorMetadata(NamedTuple):
     is_quantized : bool
     qparams: Dict[str, Any]
 
-def _extract_tensor_metadata(result : torch.Tensor, include_contiguity=True) -> TensorMetadata:
+    # Sparse metadata when layout in [ torch.sparse_coo/csr/csc/bsr/bsc ]
+    #   batch_dim + sparse_dim + dense_dim = ndim = len(shape)
+    #   blocksize for bsr/bsc
+    #   idx_dtype denote the types used for compressed indices and positions
+    batch_dim : Optional[int]
+    sparse_dim : Optional[int]
+    dense_dim : Optional[int]
+    blocksize : Optional[Tuple[int, int]]
+    idx_dtype : Optional[torch.dtype]
+
+
+def _extract_sparse_tensor_metadata(
+    t: torch.Tensor,
+) -> Tuple[int, int, int, Optional[Tuple[int, int]], Optional[torch.dtype]]:
+    # Set index type.
+    if t.layout is torch.sparse_coo:
+        idx_dtype = t.indices().dtype
```

Comment:
```suggestion
        idx_dtype = t._indices().dtype
```
to support uncoalesced COO tensors.